### PR TITLE
Test two combinations in triangular test sets

### PR DESCRIPTION
**Pull request by jishnub**
Change `elty1` to `elty2` to avoid repeating the condition.

### CODE REVIEW

**Review by dkarrasch — APPROVED** [review]
Does that reduce the number of cases and hence test runtime?

**Comment by jishnub** [comment]
Yes, this reduces the number of combinations being tested. After this, tests will run for only two values of `elty2` when `elty1` is not `Int` or `ComplexF32`.

**Comment by jishnub** [comment]
Actually, this increases the number of cases. Currently, e.g. for `elty1 = Float64`, the only combination being tested is `(elty1, elty2) = (Float64, Float64)`, as the condition `elty1 ∈ (ComplexF32, Int)` is not satisfied. This PR would test other cases as well, with the cases tested being
```julia
(elty1, elty2) = (Float64, Float64)
(elty1, elty2) = (Float64, ComplexF32)
(elty1, elty2) = (Float64, Int64)
```
This increases the runtime by quite a bit.

Reading the comment above, the intent appears to be to test one combination of differing `eltype`s, whereas we are testing two here. Perhaps we may only loop over one `eltype` instead of two by changing the condition here.

File: test/testtriag.jl [hunk 1]
```diff
@@ -317,7 +317,7 @@ function test_triangular(elty1_types)
             @testset for elty2 in (Float32, Float64, BigFloat, ComplexF32, ComplexF64, Complex{BigFloat}, Int)
                 # Only test methods for the same element type and a single combination of mixed element types
                 # to avoid too much compilation
-                if !(elty1 == elty2 || elty1 ∈ (ComplexF32, Int) || elty1 ∈ (ComplexF32, Int))
+                if !(elty1 == elty2 || elty1 ∈ (ComplexF32, Int) || elty2 ∈ (ComplexF32, Int))
```
Comment (dkarrasch) [inline test/testtriag.jl]: What about

```suggestion
                if !(elty1 == elty2 || elty1 ∈ (ComplexF32, Int) && elty2 ∈ (ComplexF32, Int))
```

? Without deep thinking, just a quick suggestion.

Comment (jishnub) [reply test/testtriag.jl]: This restricts the `(elty1, elty2)` combinations for `Int` and `ComplexF32` types, while the floating-point types are only tested against themselves. Perhaps we may choose random combinations to test against.

**Comment by jishnub** [comment]
The current approach ensures that exactly two combinations are tested for each type. This changes the current behaviour of testing all combinations for `Int` and `ComplexF32`. This reduction in the number of combinations being tested reduces the runtime by several minutes (around 10 minutes, looking at the test run times).

**Comment by jishnub** [comment]
@andreasnoack since you were involved in splitting the test sets, do you think this approach seems reasonable? In this PR, we test each type against itself, and another type that is chosen at random.

File: test/testtriag.jl [hunk 2]
```diff
@@ -314,12 +323,10 @@ function test_triangular(elty1_types)
             @test ((A1 \ A1)::t1) ≈ M1 \ M1
 
             # Begin loop for second Triangular matrix
-            @testset for elty2 in (Float32, Float64, BigFloat, ComplexF32, ComplexF64, Complex{BigFloat}, Int)
-                # Only test methods for the same element type and a single combination of mixed element types
-                # to avoid too much compilation
-                if !(elty1 == elty2 || elty1 ∈ (ComplexF32, Int) || elty1 ∈ (ComplexF32, Int))
```
Comment (andreasnoack) [inline test/testtriag.jl:320]: How much of a difference does it make to just fix the bug here, i.e. to change the last `elty1` to `elty2`?

I'm not sure it is a good idea to test against a random eltype. It can be quite confusing with randomness in tests.

Comment (jishnub) [reply test/testtriag.jl:320]: The seed is fixed, so the tests are fully reproducible. The randomness only ensures that different types are chosen in the tests.

Changing the `elty1` to `elty2` here still leads to `Int` being tested against every type, and also tests the floating-point types against `Int` and `ComplexF32`. This increases the test runtime by quite a bit. This PR changes this to test each type against only two types: itself, and one other type.

Comment (jishnub) [reply test/testtriag.jl:320]: Please see https://github.com/JuliaLang/LinearAlgebra.jl/pull/1294#issuecomment-2818682552 for details

Comment (andreasnoack) [reply test/testtriag.jl:320]: > Changing the elty1 to elty2 here still leads to Int being tested against every type

That is because I wrote the condition incorrectly. It doesn't match the comment above it. The condition should be
```julia
if !(elty1 == elty2 || (elty1 ∈ (ComplexF32, Int) && elty2 ∈ (ComplexF32, Int)))
```

> The seed is fixed, so the tests are fully reproducible. The randomness only ensures that different types are chosen in the tests.

I see. I think fixing the conditions as above would be better. The idea is that the combination of `Int` and `ComplexF32` reaches most corner cases.

Comment (jishnub) [reply test/testtriag.jl:320]: Oh I see. I would also suggest including `BigFloat`, as it's not a bits type. But otherwise this approach should be fine, and I'll change the PR to this.

Comment (jishnub) [reply test/testtriag.jl:320]: Oh wait, doesn't this mean that we test against every `elty2` for `elty1 == Int`?

File: test/testtriag.jl [hunk 3]
```diff
@@ -314,10 +314,10 @@ function test_triangular(elty1_types)
             @test ((A1 \ A1)::t1) ≈ M1 \ M1
 
             # Begin loop for second Triangular matrix
-            @testset for elty2 in (Float32, Float64, BigFloat, ComplexF32, ComplexF64, Complex{BigFloat}, Int)
+            @testset for elty2 in push!(Set((ComplexF32, Int)), elty1)
```
Comment (andreasnoack) [inline test/testtriag.jl:317]: Why not just
```julia
elty2 in (ComplexF32, Int, elty1)
```

Comment (jishnub) [reply test/testtriag.jl:317]: This is to avoid duplicates

**Review by andreasnoack — APPROVED** [review]
